### PR TITLE
remove cloudinary provider module

### PR DESCRIPTION
**Pull request by dbauszus-glx**
The cloudinary provider module which requires the image/document payload to be passed through the xyz host has been removed.

The signer module has now been reviewed and documented.

The cloduinary signer module has now been reviewed and documented.

The module will not fail silently but return an error message.

The cloudinary entry is now using the cloudinary signer module.

The cloudinary entry is now reviewed and documented.

The cloudinary entry requires a cloudinary account. However, no third party module is required to sign requests, hence the cloudinary signer is included in the XYZ core and the cloudinary entry method is included in the mapp location entries.

### CODE REVIEW

**Review by cityremade — APPROVED** [review]
Works as expected.
I've replaced alerts and confirms with ui elements plus more human wording on removal confirm;

**Review by RobAndrewHurst — CHANGES_REQUESTED** [review]
image

when removing a document that uses cloudinary the alert says I am going to remove an image which doesn't align with the action.

We should change this to something more fitting for documents

**Review by RobAndrewHurst — CHANGES_REQUESTED** [review]
image

All of these dictionary entries except the remove_item_confirm are not used anywhere in the core, and should be removed.

**Comment by RobAndrewHurst** [comment]
> Works as expected. I've replaced alerts and confirms with ui elements plus more human wording on removal confirm;

This can't work with both images and documents.

File: lib/ui/locations/entries/cloudinary.mjs [hunk 1]
```diff
@@ -207,85 +251,123 @@ function imageLoad(e, entry) {
...
+    body: data
   })
 
   if (!response || response.error) {
-    const errorDetail = response?.error?.message ? ` Error: ${response.error.message}` : '';
-    const errorMessage = `${mapp.dictionary.document_upload_failed}${errorDetail}`;
-    alert(errorMessage);
+    const errorDetail = response?.error?.message? `Error: ${response.error.message}` : '';
+
+    alert(`Cloudinary document upload failed! ${errorDetail}`);
```
Comment (RobAndrewHurst) [inline lib/ui/locations/entries/cloudinary.mjs:331]: We will also want to make sure that all alerts are now using the core defined alerts in the module!
```suggestion
    alert(`Cloudinary document upload failed! ${errorDetail}`);
```mapp.ui.elements.alert();

**Review by RobAndrewHurst — APPROVED** [review]
I have have replaced the remove_item_confrim entry on the dictionary with two items. One for images and the other for documents.

They are then passed to the trash method as params.

**Review by AlexanderGeere — CHANGES_REQUESTED** [review]
Writing tests for the module

**Review by AlexanderGeere — APPROVED** [review]

**Review by RobAndrewHurst — APPROVED** [review]
Happy with this!

**Pull request merged**